crates/common/src/merkle_utils.rs

```diff
@@ -26,3 +28,86 @@ pub fn calculate_merkle_root(leaves: Vec<H256>) -> Result<H256, Error> {
     }
     Ok(*tree.root())
 }
+
+pub struct MergeH256;
+
+impl Merge for MergeH256 {
+    type Item = H256;
+    fn merge(left: &Self::Item, right: &Self::Item) -> Self::Item {
+        let mut hash = [0u8; 32];
+        let mut blake2b = new_blake2b();
+
+        blake2b.update(left.as_slice());
+        blake2b.update(right.as_slice());
+        blake2b.finalize(&mut hash);
+        hash.into()
+    }
+}
+
+pub type CBMT = ExCBMT<H256, MergeH256>;
+pub type CBMTMerkleProof = ExMerkleProof<H256, MergeH256>;
+
+pub fn calculate_ckb_merkle_root(leaves: Vec<H256>) -> Result<H256, Error> {
+    let tree = CBMT::build_merkle_tree(&leaves);
+    Ok(tree.root())
+}
+
+pub fn byte32_dual_hasher(arr1: &[u8], arr2: &[u8]) -> H256 {
```

Comment:
`pub fn ckb_merkle_leaf_hash(index: u32, item_hash: &[u8]) -> H256 {`

The name `byte32_dual_hasher` doesn't make sense.